nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/FlowResource.java

```diff
@@ -428,7 +429,11 @@ public Response getFlowMetrics(
             @ApiParam(
                     value = "Regular Expression Pattern to be applied against the sample label value field"
             )
-            @QueryParam("sampleLabelValue") final String sampleLabelValue
+            @QueryParam("sampleLabelValue") final String sampleLabelValue,
+            @ApiParam(
+                    value = "Name of the first field of JSON object. Applicable for JSON producer only."
+            )
+            @QueryParam("firstFieldName") final String firstFieldName
```

Comment:
What do you think about naming this parameter `rootFieldName` or `parentFieldName`?  That might help clarify the fact that not only is the field first, but it wraps all other values.

Reply:
Renamed it to rootFieldName.